crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4379,16 +4376,13 @@ impl<'a> Checker<'a> {
             .scopes
             .ancestors(scope_id)
             .skip(1)
+            .filter(|scope| scope.kind.is_function() || scope.kind.is_module())
```

Comment:
Definitions within a class aren't actually shadowed by (e.g.) definitions within a method of that class.